WordPress/src/main/java/org/wordpress/android/ui/stats/StatsAbstractFragment.java

```diff
@@ -121,18 +143,123 @@ public void onCreate(Bundle savedInstanceState) {
         AppLog.d(AppLog.T.STATS, "mStatsTimeframe: " + mStatsTimeframe.getLabel());
         AppLog.d(AppLog.T.STATS, "mDate: " + mDate); */
+
         outState.putString(ARGS_SELECTED_DATE, mDate);
         outState.putSerializable(ARGS_TIMEFRAME, mStatsTimeframe);
+        saveStatsData(outState); // Each fragment will override this
         super.onSaveInstanceState(outState);
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        // Init the UI
+        if (hasDataAvailable()) {
+            updateUI();
+        } else {
+            showPlaceholderUI();
+            refreshStats();
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        EventBus.getDefault().unregister(this);
+        super.onStop();
+    }
+
+    public boolean shouldUpdateFragmentOnUpdateEvent(StatsEvents.SectionUpdatedAbstract event) {
+        if (!isAdded()) {
+            return false;
+        }
+
+        if (!getDate().equals(event.mDate)) {
+            return false;
+        }
+
+        if (!isSameBlog(event)) {
+            return false;
+        }
+
+        if (event.mTimeframe != getTimeframe()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    boolean isSameBlog(StatsEvents.SectionUpdatedAbstract event) {
+        final Blog currentBlog = WordPress.getBlog(getLocalTableBlogID());
+        if (currentBlog != null && currentBlog.getDotComBlogId() != null) {
+            return event.mRequestBlogId.equals(currentBlog.getDotComBlogId());
+        }
+        return false;
+    }
+
+    protected void showErrorUI(VolleyError error) {
+        if (!isAdded()) {
+            return;
+        }
+
+        String label = "<b>" + getString(R.string.error_refresh_stats) + "</b>";
+
+        if (error instanceof NoConnectionError) {
+            label += "<br/>" + getString(R.string.no_network_message);
+        }
+
+        if (StatsUtils.isRESTDisabledError(error)) {
+            label += "<br/>" + getString(R.string.stats_enable_rest_api_in_jetpack);
+        }
+
+        showErrorUI(label);
+    }
+
+    protected void showErrorUI() {
+        String label = "<b>" + getString(R.string.error_refresh_stats) + "</b>";
+        showErrorUI(label);
+    }
+
+    public boolean shouldUpdateFragmentOnErrorEvent(StatsEvents.SectionUpdateError errorEvent) {
+        if (!shouldUpdateFragmentOnUpdateEvent(errorEvent)) {
+            return false;
+        }
+
+        StatsService.StatsEndpointsEnum sectionToUpdate = errorEvent.mEndPointName;
+        StatsService.StatsEndpointsEnum[] sectionsToUpdate = getSectionsToUpdate();
+        int indexOfDatamodelMatch = -1;
+        for (int i = 0; i < getSectionsToUpdate().length; i++) {
+            if (sectionToUpdate == sectionsToUpdate[i]) {
+                indexOfDatamodelMatch = i;
```

Comment:
It looks like `indexOfDatamodelMatch` is unused after the loop. You can simply return true here and return false after the loop.